src/server/v2.0/handler/ldap.go

```diff
@@ -0,0 +1,96 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/goharbor/harbor/src/common/rbac"
+	"github.com/goharbor/harbor/src/controller/ldap"
+	"github.com/goharbor/harbor/src/lib/errors"
+	ldapModel "github.com/goharbor/harbor/src/pkg/ldap/model"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/ldap"
+)
+
+type ldapAPI struct {
+	BaseAPI
+	ctl ldap.Controller
+}
+
+func newLdapAPI() *ldapAPI {
+	return &ldapAPI{ctl: ldap.Ctl}
+}
+
+func (l *ldapAPI) PingLdap(ctx context.Context, params operation.PingLdapParams) middleware.Responder {
+	if err := l.RequireSystemAccess(ctx, rbac.ActionRead, rbac.ResourceConfiguration); err != nil {
+		return l.SendError(ctx, err)
+	}
+	basicCfg := ldapModel.LdapConf{
+		URL:            params.Ldapconf.LdapURL,
+		BaseDn:         params.Ldapconf.LdapBaseDn,
+		SearchDn:       params.Ldapconf.LdapSearchDn,
+		Filter:         params.Ldapconf.LdapFilter,
+		SearchPassword: params.Ldapconf.LdapSearchPassword,
+		UID:            params.Ldapconf.LdapUID,
+		Scope:          int(params.Ldapconf.LdapScope),
+		VerifyCert:     params.Ldapconf.LdapVerifyCert,
+	}
+	payload := &models.LdapPingResult{}
+	suc, err := l.ctl.Ping(ctx, basicCfg)
+	payload.Success = suc
+	if err != nil {
+		payload.Message = fmt.Sprintf("error: %v", err)
+	}
+	return operation.NewPingLdapOK().WithPayload(payload)
+}
+
+func (l *ldapAPI) SearchLdapUser(ctx context.Context, params operation.SearchLdapUserParams) middleware.Responder {
+	if err := l.RequireSystemAccess(ctx, rbac.ActionRead, rbac.ResourceLdapUser); err != nil {
```

Comment:
It's better to use `rbac.ActionList` than the `rbac.ActionRead` because this API returns an array of LDAP users.

`rbac.ActionRead` is used when the API returns only one instance, eg read a specific project.